src/common/src/system_param/local_manager.rs

```diff
@@ -40,9 +41,23 @@ pub struct LocalSystemParamsManager {
 }
 
 impl LocalSystemParamsManager {
-    pub fn new(params: SystemParamsReader) -> Self {
-        let params = Arc::new(ArcSwap::from_pointee(params));
+    pub fn new(initial_params: SystemParamsReader) -> Self {
+        let params = Arc::new(ArcSwap::from_pointee(initial_params.clone()));
         let (tx, _) = channel(params.clone());
+
+        // Spawn a task to run the common handler.
+        tokio::spawn({
+            let mut rx = tx.subscribe();
+            async move {
+                let handler = CommonHandler::new(initial_params);
+
+                while rx.changed().await.is_ok() {
+                    let new_params = (**rx.borrow_and_update().load()).clone();
+                    handler.handle_change(new_params);
```

Comment:
Why do we need this in 3 places

Reply:
The code paths for subscribing to system parameter changes on meta nodes and other nodes are completely different. 😄 Further refactoring is needed.